apps/meteor/client/components/Omnichannel/hooks/useDepartmentsList.spec.ts

```diff
@@ -0,0 +1,107 @@
+import { mockAppRoot } from '@rocket.chat/mock-providers';
+import { renderHook, waitFor } from '@testing-library/react';
+
+import { useDepartmentsList } from './useDepartmentsList';
+
+const initialDepartmentsListMock = Array.from(Array(25)).map((_, index) => {
+	return {
+		_id: `${index}`,
+		name: `test_department_${index}`,
+		enabled: true,
+		email: `test${index}@email.com`,
+		showOnRegistration: false,
+		showOnOfflineForm: false,
+		type: 'd',
+		_updatedAt: '2024-09-26T20:05:31.330Z',
+		offlineMessageChannelName: '',
+		numAgents: 0,
+		ancestors: undefined,
+		parentId: undefined,
+	};
+});
+
+it('should not add selected department if it is already in the departments list on first fetch', async () => {
+	const selectedDepartmentMappedToOption = {
+		_id: '5',
+		label: 'test_department_5',
+		value: '5',
+	};
+
+	const { result } = renderHook(
+		() =>
+			useDepartmentsList({
+				filter: '',
+				onlyMyDepartments: true,
+				haveAll: true,
+				showArchived: true,
+				selectedDepartment: '5',
+			}),
+		{
+			legacyRoot: true,
+			wrapper: mockAppRoot()
+				.withEndpoint('GET', '/v1/livechat/department', () => ({
+					count: 25,
+					offset: 0,
+					total: 25,
+					departments: initialDepartmentsListMock,
+				}))
+				.build(),
+		},
+	);
+
+	await waitFor(() => expect(result.current.itemsList.items).toContainEqual(selectedDepartmentMappedToOption));
+	// The expected length is 26 because the hook will add the 'All' item on run time
+	await waitFor(() => expect(result.current.itemsList.items.length).toBe(26));
+});
```

Comment:
I believe this test case could be improved by mocking the endpoint `/v1/livechat/department/:_id` and making sure it is not being called. This can be achieved with a combination of `withEndpoint` and `jest.fn()`. If you need help implementing, give me a heads up.